language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToFunctionCodeAction.java

```diff
@@ -403,6 +459,45 @@ private String getFunctionName(CodeActionContext context, NonTerminalNode matche
         return NameUtil.generateTypeName(EXTRACTED_PREFIX, visibleSymbolNames);
     }
 
+    private List<NonTerminalNode> getPossibleExpressionNodes(NonTerminalNode node,
+                                                             ExpressionNodeValidator nodeValidator) {
```

Comment:
Is there a real usage of passing the `ExpressionNodeValidator` here? We can create a node validator inside this method and use it. Node validator doesn't take any constructor args making it context insensitive.